oak_runtime/src/node/http/tests.rs

```diff
@@ -244,12 +273,30 @@ async fn send_request(
     let client: hyper::client::Client<_, hyper::Body> =
         hyper::client::Client::builder().build(https);
 
-    let request = hyper::Request::builder()
-        .method(http::Method::GET)
-        .uri(uri)
-        .header(oak_abi::OAK_LABEL_HTTP_PROTOBUF_KEY, label_bytes)
-        .body(hyper::Body::empty())
-        .unwrap();
-
-    client.request(request).await
+    // The client thread may start sending the requests before the server is up. In this case, the
+    // request will be rejected with a "ConnectError". To make the tests are stable, we need to
+    // retry sending the requests until the server is up. To distinguish between these cases and
+    // actual errors (e.g., errors due to invalid TLS certificates), we need to check the cause of
+    // the error.
+    loop {
+        let request = hyper::Request::builder()
+            .method(http::Method::GET)
+            .uri(uri)
+            .header(oak_abi::OAK_LABEL_HTTP_PROTOBUF_KEY, label_bytes.clone())
+            .body(hyper::Body::empty())
+            .unwrap();
+
+        match client.request(request).await {
+            Ok(reps) => return Ok(reps),
+            Err(error) => {
+                // We cannot access the cause of the error, so we need to check the string instead.
+                let error_str = format!("{:?}", error);
+                if error_str.contains("ConnectError") {
```

Comment:
Could you add a link to https://github.com/hyperium/hyper/blob/6e7e4e2cd50197fe2edda22d1646b89b1f6513b3/src/client/connect/http.rs#L425-L436 or wherever this string is coming from, so that if things start failing in the future because of some upstream change, we know how to fix it?

Reply:
Done.